src/test/java/org/sagebionetworks/bridge/spring/controllers/CRCControllerTest.java

```diff
@@ -0,0 +1,669 @@
+    
+    @Captor
+    ArgumentCaptor<ReportData> reportCaptor;
+    
+    @Captor
+    ArgumentCaptor<Account> accountCaptor;
+    
+    App app;
+    
+    Account account;
+    
+    @InjectMocks
+    @Spy
+    CRCController controller;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        MockitoAnnotations.initMocks(this);
+
+        account = Account.create();
+        account.setHealthCode(HEALTH_CODE);
+        account.setAccountSubstudies(ImmutableSet.of(ACCT_SUB1, ACCT_SUB2));
+        account.setDataGroups(ImmutableSet.of("group1"));
+        
+        app = App.create();
+        when(mockAppService.getApp(APP_ID)).thenReturn(app);
+        
+        doReturn(mockRequest).when(controller).request();
+        doReturn(mockResponse).when(controller).response();
+        doReturn(TestConstants.TIMESTAMP).when(controller).getTimestamp();
+    }
+    
+    @AfterMethod
+    public void afterMethod() {
+        BridgeUtils.setRequestContext(RequestContext.NULL_INSTANCE);
+    }
+    
+    @Test
+    public void updateParticipantCallsExternal() throws Exception {
+        StudyParticipant participant = new StudyParticipant.Builder()
+                .withId("thisIdWillBeIgnored")
+                .withDataGroups(makeSetOf(CRCController.AccountStates.SELECTED, "group1")).build();
+        mockRequestBody(mockRequest, participant);
+        
+        UserSession session = new UserSession();
+        session.setAppId(APP_ID);
+        session.setParticipant(new StudyParticipant.Builder().withId(USER_ID).build());
+        doReturn(session).when(controller).getAuthenticatedSession(RESEARCHER);
+
+        StatusMessage message = controller.updateParticipant("targetUserId");
+        assertNotNull(message);
+
+        verify(mockParticipantService).updateParticipant(eq(app), participantCaptor.capture());
+        verify(controller).createLabOrder(any()); // expand on this
+
+        StudyParticipant captured = participantCaptor.getValue();
+        assertEquals(captured.getDataGroups(), makeSetOf(CRCController.AccountStates.TESTS_REQUESTED, "group1"));
+        assertEquals(captured.getId(), "targetUserId");
+    }
+    
+    @Test
+    public void updateParticipantDoesNotCallExternal() throws Exception {
+        StudyParticipant participant = new StudyParticipant.Builder()
+                .withDataGroups(makeSetOf(CRCController.AccountStates.DECLINED, "group1")).build();
+        mockRequestBody(mockRequest, participant);
+        
+        UserSession session = new UserSession();
+        session.setAppId(APP_ID);
+        session.setParticipant(new StudyParticipant.Builder().withId(USER_ID).build());
+        doReturn(session).when(controller).getAuthenticatedSession(RESEARCHER);
+
+        when(mockParticipantService.getParticipant(app, USER_ID, true))
+                .thenReturn(new StudyParticipant.Builder().build());
+        
+        controller.updateParticipant(USER_ID);
```

Comment:
Should also use a different user ID in this test as well.

Reply:
Changed to use different user ID.